### PR TITLE
Create the initial data pipeline task definition

### LINKED ISSUE #284: Implementation for the hook to start the pipeline
Depends on #283.

After we have the hook, we have to implement its logic.

The logic will be kind of a decision task (similar to https://github.com/marco-c/breakpad-win-update-symbols/blob/3af1618c61f57f3ff87bf5f68e595d899316a8a2/run-taskcluster.py#L94), which will trigger a taskgraph (data retrieval -> training -> deployment of http service with trained models).

### LINKED ISSUE #291: Pipeline hook should be also triggered when a new release is made
See also #283.

### PR DESCRIPTION
**Pull request by Lothiraldan**
There is a hook (which runs every day at midnight) that will spawn the
data-pipeline using the latest docker images that were build on latest
release. The hook itself is updated on each release and is versionned in this
repository.

The hook will runs once every week and on every successful release.

Add task for building the Docker image for HTTP service.

Fixes #284 and fixes #291.

### CODE REVIEW

File: data-pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,229 @@
+version: 1
+tasks:
+  - ID: commit-retrieval
+    created: {$fromNow: ''}
+    deadline: {$fromNow: '12 hours'}
+    provisionerId: aws-provisioner-v1
+    workerType: releng-svc # TODO FIX ME BEFORE MERGING
```
Comment (marco-c) [inline data-pipeline.yml]: TODO :)

File: data-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+      maxRunTime: 43200
+      image: mozilla/bugbug-commit-retrieval
+
+      artifacts:
+        public/commits.json.xz:
+          path: /data/commits.json.xz
+          type: file
+
+    routes:
+      # - notify.email.release-mgmt-analysis@mozilla.com.on-failed"
```
Comment (marco-c) [inline data-pipeline.yml]: Let's bring this back now

Comment (marco-c) [reply data-pipeline.yml]: (here and elsewhere)

File: data-pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+      name: bugbug bugs retrieval
+      description: bugbug bugs retrieval
+      owner: release-mgmt-analysis@mozilla.com
+      source: https://github.com/mozilla/bugbug/raw/master/data-pipeline.yml
+
+  - ID: train-component
+    created: {$fromNow: ''}
+    deadline: {$fromNow: '48 hours'}
+    provisionerId: aws-provisioner-v1
+    workerType: releng-svc # TODO FIX ME BEFORE MERGING
```
Comment (marco-c) [inline data-pipeline.yml]: The train ones should use the -compute variant (releng-svc-compute)

File: infra/spawn_data_pipeline.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This script triggers the data pipeline for the bugbug project
+"""
+
+from __future__ import print_function
```
Comment (marco-c) [inline infra/spawn_data_pipeline.py]: We probably don't need this with Python3

File: infra/spawn_data_pipeline.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    tasks = []
+
+    with open(args.data_pipeline_json) as pipeline_file:
+        raw_tasks = yaml.load(pipeline_file.read())
+
+    for task in raw_tasks["tasks"]:
+        # Try render the task template
+        context = {}
+        payload = jsone.render(task, context)
+        # raise Exception(payload)
```
Comment (marco-c) [inline infra/spawn_data_pipeline.py]: You can remove this

File: infra/spawn_data_pipeline.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        # # TODO: the task_id will not match the taskId in the yaml file
```
Comment (marco-c) [inline infra/spawn_data_pipeline.py]: What were you planning to do here?

File: infra/spawn_data_pipeline.py [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+        task_id = taskcluster.utils.slugId()
+        task_internal_id = payload.pop("ID")
+
+        if task_internal_id in id_mapping:
+            raise ValueError("Conflicting IDS {}".format(task_internal_id))
```
Comment (marco-c) [inline infra/spawn_data_pipeline.py]: Nit: IDs

File: infra/taskcluster-hook-pipeline-start.json [hunk 8]
```diff
@@ -29,25 +35,27 @@
...
             },
-            "image": "hello-world",
+            "image": "python:3.7.3",
             "maxRunTime": 7200
         },
         "priority": "normal",
         "provisionerId": "aws-provisioner-v1",
         "retries": 5,
         "routes": [
-            "notify.email.release-mgmt-analysis@mozilla.com.on-failed",
```
Comment (marco-c) [inline infra/taskcluster-hook-pipeline-start.json]: Let's bring this back

File: infra/spawn_data_pipeline.py (hunk 6 above)
Comment (Lothiraldan) [reply infra/spawn_data_pipeline.py]: I forget to cleanup the TODO when I introduced the internal ID mapping

**Pull request merged**